app/src/main/java/com/example/wishlist/adapters/ShoppingCartAdapter.kt

```diff
@@ -25,10 +28,15 @@ class ShoppingCartAdapter(var cartItems: List<Product> = emptyList()): RecyclerV
                 .into(productImage)
         }
     }
+}
 
-    override fun getItemCount(): Int {
-        return cartItems.size
-    }
+class CartItemDiffCallback(): DiffUtil.ItemCallback<Product>() {
 
+    override fun areItemsTheSame(oldItem: Product, newItem: Product): Boolean {
+        return oldItem.id == newItem.id
+    }
 
+    override fun areContentsTheSame(oldItem: Product, newItem: Product): Boolean {
+        return oldItem == newItem
```

Comment:
Does this check all attributes of the product?

Reply:
Tested the evaluation above and it does check all the attributes